GVFS.sln

```diff
@@ -201,243 +217,385 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "GVFS.PostIndexChangedHook.W
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug.Linux|x64 = Debug.Linux|x64

```

Comment:
The good news is I was able to build your branch on Mac with `Scripts/Mac/BuildGVFSForMac.sh`.  The bad news is that something is configured in a way that VS for Mac doesn't like, but that can get sorted out before this is merged to master.

Reply:
Hm, can you describe it at all? I just tried a reasonably fresh build with VS for Mac of this tree and everything worked OK.

I trashed `BuildOutput`, `packages`, `Publish`, ran enough of `BuildGVFSForMac.sh` to generate the constants and assembly version files, then built the solution. Everything looks fine.

---

I will note that I encounter issues regularly if I try to build with VS and then build script (or vice versa) without cleaning everything in between — VS gets confused, red-underlines everything, says `System` is undefined, etc. etc. Just to see if I could trigger it in this case, after building with VS per above, I ran the full `BuildGVFSForMac.sh`, and then again in VS, and this _one_ time it actually works.

(Well, actually, I get the incompatible SDK error thing on two projects, but that's not what I'm talking about and is already known!)

Reply:
Hi, these are the errors I'm seeing:

> Error: NuGet packages need to be restored before building. NuGet MSBuild targets are missing and are needed for building. The NuGet MSBuild targets are generated when the NuGet packages are restored. (GVFS.Common)
>
> Error: NuGet packages need to be restored before building. NuGet MSBuild targets are missing and are needed for building. The NuGet MSBuild targets are generated when the NuGet packages are restored. (GVFS.Hooks.Mac)
>
> Error: NuGet packages need to be restored before building. NuGet MSBuild targets are missing and are needed for building. The NuGet MSBuild targets are generated when the NuGet packages are restored. (GVFS.Tests)

I tried deleting BuildOutput, packages, and Publish but it didn't help.

Reply:
Trying to make my steps reproducible:
* Start with VS closed.
* Deleted `BuildOutput`, `packages`, `Publish`. The clone is at `~/microsoft/VFSForGit/src`, and the `VFSForGit` directory is empty except for the clone at `src`.
* Performed `git clean -fdx` in the clone.
* Opened `GVFS.sln` in Visual Studio for Mac (Community, 8.0.4 build 0).
* It automatically restored packages when opening the solution. (I'll omit the entire Package Console log, but the last line reads `Packages successfully restored.`)
* Tried to build the solution. Failed because I have `Debug.Windows` selected and a bunch of "This project type is not supported by Visual Studio Community 2019 for Mac." messages appeared.
* Switched to `Debug.Mac`. Failed because `GVFSConstants.GitVersion.cs` is not found.
* Made [these changes](https://gist.github.com/kivikakk/bbc1a0ca7d43ead9962eca3a0ebb8ba5) to `BuildGVFSForMac.sh` and ran the script to create the two generated `.cs` files.
* Built the solution again.
* ```
  ========== Build: 16 succeeded, 0 failed, 2 up-to-date, 7 skipped ==========

  Build successful.
  ```

It's possible the `git clean` step was necessary? I'm not sure.

Reply:
Ahh thanks, I did forget to run `git clean`, let me give that a try.